web/src/echo.ts

```diff
@@ -162,54 +240,59 @@ export function build_display_recipient(message) {
             id: message.sender_id,
             email: message.sender_email,
             full_name: message.sender_full_name,
+            is_mirror_dummy: false,
         });
     }
     return display_recipient;
 }
 
-export function insert_local_message(message_request, local_id_float, insert_new_messages) {
+export function insert_local_message(
+    message_request: MessageRequest,
+    local_id_float: number,
+    insert_new_messages: (messages: LocalMessage[], send_by_this_client: boolean) => Message[],
+): Message {
     // Shallow clone of message request object that is turned into something suitable
     // for zulip.js:add_message
     // Keep this in sync with changes to compose.create_message_object
-    let message = {...message_request};
-
-    message.raw_content = message.content;
-
-    // NOTE: This will parse synchronously. We're not using the async pipeline
-    message = {
-        ...message,
-        ...markdown.render(message.raw_content),
+    const raw_content = message_request.content;
+    const topic = message_request.topic;
+
+    const local_message: LocalMessage = {
+        ...message_request,
+        ...markdown.render(raw_content),
+        raw_content,
+        content_type: "text/html",
+        sender_email: people.my_current_email(),
+        sender_full_name: people.my_full_name(),
+        avatar_url: current_user.avatar_url,
+        timestamp: Date.now() / 1000,
+        local_id: local_id_float.toString(),
+        locally_echoed: true,
+        id: local_id_float,
+        resend: false,
+        is_me_message: false,
+        topic_links: topic !== undefined ? markdown.get_topic_links(topic) : [],
```

Comment:
I think the topic will never be `undefined` here instead it will be an empty string in case we are sending a private message. Hence `topic !== undefined` will always be true. I guess the intention here is to check if the `topic.length >0`.

Reply:
Yeah the actually should have been for falsy (empty string). Thanks for catching that. It was a potential bug fixed in https://github.com/zulip/zulip/pull/30702/commits/7570fbddca68e27a8654a40eff39deb1aa76206e